Buff scythes, goats, and plantbgone vs PLANT biotypes

timothymtorres opened the pull request:
## About The Pull Request
This buffs scythes, goats, and plantbgone vs PLANT biotypes:

- Scythes now deal x1.5 damage to venus flytraps (3 hits to kill)
- Scythes now target the flower bud vines
- Goats now target flower bud vines and deal 15 damage to PLANT biotypes
- Goats have a eating sound whenever they bite PLANT biotypes
- Plantbgone now does 2 dmg per unit to PLANT biotypes (10 dmg per spray)
- Plantbgone now has a 75% chance to remove weeds and deals large damage to flower buds
- Weed control crates now come with a pair of leather gloves
- Golems are immune to thorn effects
- Any kind of thick glove material will prevent thorn effects when attacking
- Flower buds will now take x4 damage from fire and sharp weapons (unless they have fire trait)
- Regular scythes are now a sharp object

Also this fixes a few runtimes with spacevines and nulls.  The bane element now accepts `mob_biotypes` bitflags as an argument.

## Why It's Good For The Game
Before my changes:

- Plant-b-gone was doing 0.4 dmg per unit to PLANT biotypes (2 dmg per spray)
- Scythes took 5 hits to kill venus flytraps
- Goats only affected podpeople
- Flower bud vines were being ignored by weed killing code
- Plantbgone only had a 50% chance to remove weeds (and this was very inconsistent due to RNG)
- Botanical gloves and thick gloves didn't protect from thorns
- Golems were getting pierced by thorns despite having pierce immunity
- Flower buds were not taking the x4 damage like they should have been
- Regular scythes were not a sharp object, but other scythes (chaplain's, megafauna loot) were sharp

This makes the weed killer crate more effective since people were complaining about it being worthless vs vines and flower buds.  These changes give people more options to respond to threats vs plants.  

## Changelog
:cl:
add: Add a pair of leather gloves to weed control crate
balance: Mobs with the PLANT biotypes (venus flytraps, pod people, killer tomatoes) are now much weaker vs scythes, goats, and plantbgone.
balance: Plantbgone is now more effective at destroying weeds.
balance: Regular scythes are now a sharp object
fix: Fixed scythes, goats, and plantbgone not affecting flower bud vines.
fix: Thick and botanical gloves not protecting from thorns
fix: Golems not having pierce immunity from thorns
fix: Runtime where vines tried to spread into null turf
fix: Runtime where null vines that were destroyed were trying to spread to nearby turfs
soundadd: Add eat food sound when goats eat plants
code: Improved goat targeting code
code: The bane element now accepts `mob_biotypes` bitflags as an argument.
/:cl:

timothymtorres committed:
Fix mobs moving inside containers

timothymtorres committed:
Add damage to venus flytraps with scythes

timothymtorres committed:
Buff plantbgone dmg effect vs PLANT biotypes

timothymtorres committed:
Buff goat dmg effect vs PLANT biotype

Shadowflame909 commented:
Another Era in the great kudzu war. Im excited to see the potential future kudzu buff from this. I hope it's entertaining and not just blob 0.5

jughu commented:
Make the bulbs fragile to sharp/fire weapons in general, in the past they were always fragile and as trade off for this fragility they spawned super strong mobs

the-orange-cow commented:
> Make the bulbs fragile to sharp/fire weapons in general, in the past they were always fragile and as trade off for this fragility they spawned super strong mobs

Maneaters aren't super strong though.  They're pretty fragile simplemobs whose only real benefit is a knockdown with a cooldown.

necromanceranne commented:
Tim did you test to see if scythes were actually cutting multiple vines? I believe Lukas was saying how this wasn't currently working.

improvedname commented:
> > Make the bulbs fragile to sharp/fire weapons in general, in the past they were always fragile and as trade off for this fragility they spawned super strong mobs
> 
> Maneaters aren't super strong though. They're pretty fragile simplemobs whose only real benefit is a knockdown with a cooldown.

you mean the fragile weak simplemob which heal at every attack and where you have to need a strong weapon otherwise it just outheals you?

timothymtorres committed:
Merge branch 'patch-2' of https://github.com/timothymtorres/tgstation into patch-2

timothymtorres committed:
Remove old branch code

timothymtorres committed:
Fix identical turf name

timothymtorres committed:
Fix target to use correct health proc

timothymtorres committed:
Fix runtime due to null type

timothymtorres committed:
Fix runtime in growth_queue using null

So after a vine gets destroyed with a grenade or something, it gets qdel but I don't believe it ever gets removed from growth_queue so there is a null ref there.

timothymtorres committed:
Buff plantbgone to have 75% to remove weeds

timothymtorres committed:
Fix burn and sharpness damage multipliers for flower buds

timothymtorres committed:
Change goats to deal large damage to flower buds

timothymtorres committed:
Change thorny vines to ignore golems and thick gloves

timothymtorres committed:
Add leather gloves to weed control crate

timothymtorres committed:
Fix botanical gloves not providing protection

timothymtorres committed:
Add the bane element for the other scythes

timothymtorres commented:
> Tim did you test to see if scythes were actually cutting multiple vines? I believe Lukas was saying how this wasn't currently working.

Yes.  If the vine is woodened/hardended then it will reduce damage from sharp weapons 50%

timothymtorres committed:
Merge remote-tracking branch 'upstream/master' into patch-2

timothymtorres committed:
Fix bad user vars

timothymtorres committed:
More BALANCE

timothymtorres committed:
Change sharp weapons to deal x4 damage

timothymtorres committed:
Refactor if loop to be simple

timothymtorres committed:
Balance the bane element to not be overwhelming

timothymtorres committed:
Fix plantbgone ignoring flower buds

tralezab reviewed (changes requested):
2 things in reality

timothymtorres committed:
Refactor bane element to accept mob biotypes

timothymtorres commented on code/game/objects/items/holy_weapons.dm:
```diff
@@ -253,7 +253,7 @@
 	effectiveness = 110, \
 	)
-	//the harvest gives a high bonus chance
+	AddElement(/datum/element/bane, /mob/living/simple_animal/hostile/venus_human_trap, damage_multiplier = 0.5, requires_combat_mode = FALSE)
```
Added the biotype functionality to the bane element via https://github.com/tgstation/tgstation/pull/72889/commits/7592d9fdc5d66e3fd7f711ac69eeae86529f5860

timothymtorres commented on code/modules/hydroponics/hydroitemdefines.dm:
```diff
@@ -529,6 +530,7 @@
 	speed = 9 SECONDS, \
 	effectiveness = 105, \
 	)
+	AddElement(/datum/element/bane, /mob/living/simple_animal/hostile/venus_human_trap, damage_multiplier = 0.5, requires_combat_mode = FALSE)
```
Added the biotype functionality to the bane element via https://github.com/tgstation/tgstation/pull/72889/commits/7592d9fdc5d66e3fd7f711ac69eeae86529f5860

timothymtorres committed:
Typecast step into turf

timothymtorres commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -61,7 +61,15 @@
 
 	for(var/direction in shuffle(list(1,2,4,8,5,6,9,10)))
 		var/step = get_step(src, direction)
```
Applied with https://github.com/tgstation/tgstation/pull/72889/commits/e71712ac86f336c65a6c18217048779f3fbe0eab

timothymtorres commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -74,29 +82,44 @@
+
+	plant_target.adjustBruteLoss(20)
+	playsound(src, 'sound/items/eatfood.ogg', rand(30, 50), TRUE)
+	var/obj/item/bodypart/edible_bodypart
```
I deliberately moved this code logic out of the human check so that it affects simple mobs that have the plant biotype.  The human check is only to dismember the limbs.  We still want to deal brute damage, play the sound, and show the message.

timothymtorres committed:
TODO

timothymtorres committed:
Fix mob biotype flags to work with bane element
